docs/BasicCommunication/OnPutFile/index.md

```diff
@@ -13,25 +13,31 @@ Purpose
 
 !!! MUST
 
-Use the appropriate uploaded file according to its workflow (IVSU, SystemRequest, RPC's). See diagrams listed below.
+1. Use the appropriate uploaded file according to its workflow (IVSU, SystemRequest, RPC's). See diagrams listed below.  
+2. Whenever HMI gets RPC with `Image` which has `.isTemplate` set to true, the HMI has to:
 
+- load the proper image pattern
+- extract alpha channel from the template image
+- apply the alpha channel to the image pattern
+- use this newly generated image instead of the uploaded one  
+
+and whenever the UI changes, the HMI has to recreate images currently visible on the screen.
 !!!
 
 !!! NOTE
 
 The list of RPCs and data structures that `OnPutFile` affects are:
 
   * Show (Image, SoftButton)
-  * ShowConstantTBT (SoftButton)
+  * ShowConstantTBT (Image, SoftButton)
   * CreateInteractionChoiceSet (Image)
-  * DeleteInteractionChoiceSet (Image)
   * SetGlobalProperties (Image, VrHelpItem)
   * ResetGlobalProperties (Image, VrHelpItem)
-  * UpdateTurnList (Turn)
+  * UpdateTurnList (Turn, SoftButton)
```

Comment:
duplicated with the last item `UpdateTurnList (SoftButton)` in this list?

Reply:
@yang1070, thank you.
fixed with https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/97/commits/04340d3bfea399bd96e16d7bfc3e0220b23f00ef